File: backend/app/schemas/health.py
from pydantic import BaseModel, Field
from datetime import datetime


class HealthResponse(BaseModel):
    status: str = Field(default="ok", description="Status of the API service")
    project_name: str = Field(..., description="Project Name")
    version: str = Field(..., description="API Version")
    timestamp: datetime = Field(default_factory=datetime.utcnow, description="Current UTC timestamp")
    backend_url: str = Field(..., description="Configured Backend URL")
    frontend_url: str = Field(..., description="Configured Frontend URL")

File: frontend/src/lib/api/health.ts
import { apiClient } from "./client";

export interface HealthResponse {
  status: string;
  project_name: string;
  version: string;
  timestamp: string;
  backend_url: string;
  frontend_url: string;
}

export async function fetchBackendHealth(): Promise<HealthResponse> {
  return apiClient<HealthResponse>("/api/v1/health");
}
